src/main/java/com/apicatalog/jsonld/flattening/NodeMap.java

```diff
@@ -47,37 +47,58 @@ public void set(String graphName, String subject, String property, JsonValue val
         }
 
         index
-            .computeIfAbsent(graphName, x -> new LinkedHashMap<>())
-            .computeIfAbsent(subject, x -> new LinkedHashMap<>())
-            .put(property, value);
+                .computeIfAbsent(graphName, x -> new LinkedHashMap<>())
+                .computeIfAbsent(subject, x -> new LinkedHashMap<>())
+                .put(property, value);
     }
 
     public JsonValue get(String graphName, String subject, String property) {
 
-        if (index.containsKey(graphName) && index.get(graphName).containsKey(subject)) {
-            return index.get(graphName).get(subject).get(property);
+        var indexGraphName = index.get(graphName);
+        if (indexGraphName != null) {
+
+            var indexSubject = indexGraphName.get(subject);
+            if (indexSubject != null) {
+                return indexSubject.get(property);
+            }
         }
 
         return null;
     }
 
     public Map<String, JsonValue> get(String graphName, String subject) {
 
-        if (index.containsKey(graphName)) {
-            return index.get(graphName).get(subject);
+        var indexGraphName = index.get(graphName);
+
+        if (indexGraphName != null) {
+            return indexGraphName.get(subject);
         }
 
         return null;
     }
 
     public boolean contains(String graphName, String subject) {
-        return index.containsKey(graphName) && index.get(graphName).containsKey(subject);
+        var indexGraphName = index.get(graphName);
+
+        if (indexGraphName != null) {
+            return indexGraphName.containsKey(subject);
+        }
+
+        return false;
+
     }
 
     public boolean contains(String graphName, String subject, String property) {
-        return index.containsKey(graphName)
-                    && index.get(graphName).containsKey(subject)
-                    && index.get(graphName).get(subject).containsKey(property);
+       var indexGraphName = index.get(graphName);
+
+        if (indexGraphName != null) {
+            var indexSubject = indexGraphName.get(subject);
+            if (indexSubject != null) {
+                return indexSubject.containsKey(property);
+            }
+        }
+        return false;
+
```

Comment:
The old code called both containsKey(...) and get(...), new code only calls get(...) and then does a null check instead. It's a very minor performance improvement.